hip-service/wwwroot/swagger.yaml

```diff
@@ -0,0 +1,172 @@
+
+tags:
+  - name: discovery
+
+paths:
+  /patients/discover/:
+    post:
+      tags:
+      - discovery
+      summary: Discover patient's accounts
+      description: >
+        Return only one patient record with (potentially masked) associated care contexts
+          1. **At least one of the verified identifier matches.**
+          2. **Filter out records using unverified, firstName, secondName, gender and dob
+          if there are more than one patient records found from step 1.**
+      consumes:
+      - application/json
+      - application/xml
+      produces:
+      - application/json
+      - application/xml
+      requestBody:
+        content:
+          application/json:
+            required: true
+            schema:
+              $ref: '#/components/requestBodies/PatientDiscoveryRequest'
+          application/xml:
+            required: true
+            schema:
+              $ref: '#/components/requestBodies/PatientDiscoveryRequest'
+      responses:
+        '200':
+          description: A patient record with one or more care contexts
+          schema:
+            $ref: '#/components/responses/PatientDiscoveryResponse'
+        '400':
+          description: >
+            **Causes:**
+              * Empty verified identifiers.
+              * Format mismatch of any of attributes.
+                | type   | Format/Allowed Values|
+                | ------- | ----------------    |
+                | dob     | yyyy-mm-dd          |
+                | gender  | male/female/unknown |
+                | phone-number| 10 digits       |
+
+          schema:
+            $ref: '#/components/responses/PatientDiscoveryErrResponse'
+        '404':
+          description: >
+            **Causes:**
+              * No matching patient record found for the given identifiers.
+              * There are more than one definitive match for a given request.
+          schema:
+            $ref: '#/components/responses/PatientDiscoveryErrResponse'
+        '500':
+          description: >
+            **Causes:**
+              * Downstream system(s) is down.
+              * Unhandled exceptions.
+          schema:
+            $ref: '#/components/responses/PatientDiscoveryErrResponse'
+
+components:
+  requestBodies:
+    PatientDiscoveryRequest:
+      type: object
+      properties:
+        verifiedIdentifiers:
+          type: array
+          items:
+            $ref: '#/components/schemas/Identifier'
+          xml:
+            name: verifiedIdentifiers
+            wrapped: true
+        unverifiedIdentifiers:
+          type: array
+          items:
+            $ref: '#/components/schemas/Identifier'
+          xml:
+            name: unverifiedIdentifiers
+            wrapped: true
+        firstName:
+          type: string
+          example: "chandler"
+        lastName:
+          type: string
+          example: "bing"
+        gender:
+          type: gender
+          enum: [male, female, unknown]
```

Comment:
India MDDS 2011, mentions M/F/T - (Male/Female/Transgender) - but this might not be very inclusive. 
I would prefer M/F/O  - but we can check with other project impl like in Financial sector

Reply:
For now, we can go ahead I guess.